SparkPost/2web2ui
# FE-1430 - Resolve y18n, is-svg, ssri, and other 2web2ui vulnerabilities

A wild @nicklemmon enters the PR!

### What Changed

#### 📦   Dependency Changes

- Removed `yargs` from package.json (not being used by us)
- Removed `cssnano` from package.json (not being used by us)
- Replaced `node-sass` with `sass` to help remove transitive dependency (and simplify our project setup by no longer relying on the node sass binary)
- `y18n` bumped by bumping `cypress-webpack-dev-server` and `optimize-css-assets-webpack-plugin`
- Manually bumps `ssri` dependency as upgrading `terser-webpack-plugin` would not resolve the issue:
  - https://github.com/npm/ssri/issues/18
  - Not to mention, [this vulnerability](https://github.com/SparkPost/2web2ui/security/dependabot/package-lock.json/ssri/open) is irrelevant I'm pretty sure? Our CSS optimization process isn't vulnerable to a DDOS attack...
- `elliptic` bumped in to the safe range by way of upgrading `webpack` to version `4.46.0`
- `yargs-parser` bumped by updating `stylelint` to the latest version
- `minimist` bumped by way of upgrading other dependencies listed above
- `braces` bumped by way of upgrading `jest-enzyme`
- The `react-dev-utils` vulnerability [does not apply to our project](https://github.com/SparkPost/2web2ui/security/dependabot/package-lock.json/react-dev-utils/open#issue_2121_link:~:text=Only%20when%20this%20function%20is%20manually,this%20issue%20does%20not%20affect%20you.)  and I recommend we close the alert as updating the package from ~7 to ~11 would be a very heavy lift (it currently breaks our dev server)

#### ✅   Other Changes

- One of these updates was causing generated CSS classnames to change, causing a snapshot test to fail - I just went ahead and removed any use of snapshots in tests for `ApiErrorBanner`
- Some snapshots appear to have been impacted by the `jest-enzyme` upgrade 😞  - need to cut a follow up ticket (or epic?) to kill Enzyme

---

### How To Test

- Run `npm install`!
- Use `npm list <package>` to verify each dependency has been upgraded to a range appropriate to resolve the relevant [security alerts](https://github.com/SparkPost/2web2ui/security/dependabot)
- Verify the `ApiErrorBanner` still renders as expected visually
- Verify the app still builds with compressed CSS correctly - `npm run build && npm run serve`
- Commit a styling linting error within a component that uses `styled-components` and try to commit it - `stylelint` should prevent commiting
- Verify CSS stemming from `.scss` CSS modules still render as expected
- Verify the dev server still works as expected by running `npm start`

### TODO

- [ ] Cut follow-up ticket to upgrade `react-dev-utils`
